### ISSUE TITLE
robot.place() block detection only based on 2 pixel

### ISSUE THREAD

**Issue by LordNocturnus** [MightyPirates/OpenComputers#2947]
OC Version: OpenComputers-MC1.12.2-1.7.2.67
MC Version: 1.12.2

while working on a placement script for 3d prints i stumbled upon the curious case that the robot most of the time could not place a 3d print(same for vanilla blocks) on top another 3d print (no other reference block and no angel block since they are broken (Issue #2573)).
I investigated a bit and found out that the robot cant place a block onto this block:
![2018-09-22_13 57 07](https://user-images.githubusercontent.com/41686329/45917074-6a125100-be70-11e8-9a98-8a660c4c861b.png)

and yes that is a block that is only missing two pixels(namely the pixels:(7,16,8) and (7,16,9) counted from the bottom left front corner (1,1,1) when placing the block)!

It gets even more interesting since the block is detected with robot.detectDown().

While this will not be a problem for most block placement programs but for the 3d-prints this is a major problem. 
For chisel and bit it is slightly different: Once one of the two pixels is air it will not place the block.

**Comment by payonel** [comment]
we basically run a ray trace to find a face of a block we can set a block to. If that ray trace runs along a line that is interrupted in an unusual way, then yes you may find that it's choice to not place a block to be unexpected. But the robot sees that you are asking it to place a block on the face of the edge of that tiny little hole, and it can't place a block there

we need to fix the place-in-air issue, and then you need to place the block in air, and not to the face of a ray-trace-detected block.

in other words, this is ~by design~ won't fix, and  #2573 needs to get fixed

**Comment by payonel** [comment]
do note that 2573 is marked for 1.7.3 and we are ACTIVELY working on 1.7.3

**Comment by LordNocturnus** [comment]
ok thanks for the fast answer

**Comment by payonel** [comment]
and thank you for working so hard to pin point the problem!
It is hard to express how grateful I am when there are bug reports where you have worked out these details of the actual points of failure. It saves me a lot of time and effort, it is quite appreciated.

And, I wanted you to know that I've moved the place-in-air bug to the top of my backlog. It'll be next up
